packages/core/src/util/initializedTooLateHeuristic.js

```diff
@@ -34,22 +40,46 @@ const patterns = [
   /\/pg\/lib\//,
   /\/pino\/lib\//,
   /\/redis\/index.js/,
+  /\/superagent\/lib\/node\/index.js/
+];
+
+const extraPatterns = [
+  // The following patterns are excluded when @contrast/agent (which also comes with the requirement to be loaded before
+  // everything else) is already loaded before @instana/core. This is to avoid false positive warnings about
+  // @instana/core being initialized too late.
+  /\/bluebird\/js\/release\//,
   /\/request\/index.js/,
-  /\/superagent\/lib\/node\/index.js/,
   /\/winston\/lib\/winston.js/
 ];
 
-let hasBeenInitializedTooLate = false;
-
 module.exports = exports = function hasThePackageBeenInitializedTooLate() {
   if (firstCall) {
     const loadedModules = Object.keys(require.cache);
-    // eslint-disable-next-line no-restricted-syntax
-    outer: for (let i = 0; i < loadedModules.length; i++) {
+
+    // pre-check (to avoid the warning in the presence of modules that can legitimately be loaded before)
+    let addExtraPatterns = true;
+    for (let i = 0; i < loadedModules.length; i++) {
+      if (/\/@contrast\/agent\//.test(loadedModules[i])) {
+        addExtraPatterns = false;
+      }
+    }
+    if (addExtraPatterns) {
+      patterns = patterns.concat(extraPatterns);
+    } else {
+      logger.debug(
+        // eslint-disable-next-line max-len
+        'Found @contrast/agent in the modules that have already been loaded. @instana/core will therefore exclude bluebird, request and winston from the check for modules that have been loaded before @instana/core.'
+      );
+    }
+
+    for (let i = 0; i < loadedModules.length; i++) {
       for (let j = 0; j < patterns.length; j++) {
         if (patterns[j].test(loadedModules[i])) {
           hasBeenInitializedTooLate = true;
-          break outer;
+          logger.debug(
```

Comment:
Instead of stopping the analysis when a problematic module has been found, continue with it and log every problematic module.